packages/front-end/components/SavedGroupForm.tsx

```diff
@@ -75,14 +79,44 @@ const SavedGroupForm: FC<{
         }))}
         helpText={current.attributeKey && "This field can not be edited."}
       />
-      <Field
-        label="Create list of comma separated values"
-        required
-        textarea
-        {...form.register("groupList")}
-      />
+      {rawTextMode ? (
+        <Field
+          containerClassName="mb-0"
+          label="Create list of comma separated values"
+          required
+          textarea
+          value={rawText}
+          onChange={(e) => {
+            setRawText(e.target.value);
+            form.setValue("groupList", e.target.value.trim().split(","));
+          }}
+        />
+      ) : (
+        <StringArrayField
+          containerClassName="mb-0"
+          label="Create list of values"
+          value={form.watch("groupList")}
+          onChange={(values) => {
+            form.setValue("groupList", values);
+            setRawText(values.join(", "));
+          }}
+          placeholder="Enter some values..."
+          delimiters={["Enter", "Tab"]}
+        />
+      )}
+      <a
+        className="d-flex flex-column align-items-end"
+        href="#"
+        style={{ fontSize: "0.8em" }}
+        onClick={(e) => {
+          e.preventDefault();
+          setRawTextMode((prev) => !prev);
+        }}
+      >
+        Switch to {rawTextMode ? "token" : "raw text"} mode
+      </a>
```

Comment:
Seems like we could pull this out into it's own component since it's being repeated ... `SavedGroupsList` or something, that would encapsulate the toggle and two types of inputs. It would expose an onChange that would always spit out a comma separated list. Not totally necessary, just a thought